pkg/oam/util/misc.go

```diff
@@ -0,0 +1,103 @@
+package util
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	plur "github.com/gertd/go-pluralize"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/oam-kubernetes-runtime/apis/core/v1alpha2"
+)
+
+const (
+	KindDeployment = "Deployment"
+	KindService    = "Service"
+
+	WorkloadChildResourcesKey = "core.oam.dev/childResourceKinds"
+)
+
+func NestedGVResource(ob map[string]interface{}) metav1.GroupVersionResource {
+	apiVersion, _, _ := unstructured.NestedString(ob, "apiVersion")
+	kind, _, _ := unstructured.NestedString(ob, "kind")
+	g, v := ApiVersion2GroupVersion(apiVersion)
+	return metav1.GroupVersionResource{
+		Group:    g,
+		Version:  v,
+		Resource: Kind2Resource(kind),
+	}
+}
+
+func ApiVersion2GroupVersion(str string) (string, string) {
+	strs := strings.Split(str, "/")
+	if len(strs) == 2 {
+		return strs[0], strs[1]
+	}
+	// core type
+	return "", strs[0]
+}
+
+func Kind2Resource(str string) string {
+	return plur.NewClient().Plural(strings.ToLower(str))
+}
+
+func Object2Unstructured(obj interface{}) (*unstructured.Unstructured, error) {
+	objMap, err := Object2Map(obj)
+	if err != nil {
+		return nil, err
+	}
+	return &unstructured.Unstructured{
+		Object: objMap,
+	}, nil
+}
+
+func Object2Map(obj interface{}) (map[string]interface{}, error) {
+	var res map[string]interface{}
+	bts, err := json.Marshal(obj)
+	if err != nil {
+		return nil, err
+	}
+	json.Unmarshal(bts, &res)
+	return res, nil
+}
+
+func FetchChildResources(ctx context.Context, r client.Reader, workload *unstructured.Unstructured) (
```

Comment:
I think something like this can live closer to the `ApplicationConfiguration` controller

Reply:
Now that AppConfig is pretty much not involved, I am not even sure it belongs to this repo.